src/stream/gst/pipeline_builder.rs

```diff
@@ -0,0 +1,263 @@
+                // have h264 or mjpg.
+                VideoEncodeType::H264 => concat!(
+                    " ! videoconvert",
+                    " ! x264enc bitrate=5000",
+                    " ! video/x-h264,profile=baseline",
+                ),
+                VideoEncodeType::MJPG => concat!(" ! jpegenc",),
+                _ => "",
+            },
+            VideoSourceType::Local(_) => match encode {
+                // Because application-rtp templates doesn't accept "YUY2", we
+                // need to transcode it. We are arbitrarily chosing the closest
+                // format available ("UYVY").
+                VideoEncodeType::YUYV => concat!(" ! videoconvert", " ! video/x-raw,format=UYVY",),
+                _ => "",
+            },
+            video_source_type => {
+                return Err(SimpleError::new(format!(
+                    "Unsupported VideoSourceType: {video_source_type:#?}.",
+                )));
+            }
+        };
+        Ok(pipeline_transcode.to_string())
+    }
+
+    fn build_pipeline_payload(
+        encode: &VideoEncodeType,
+        is_webrtcsink: bool,
+    ) -> Result<String, SimpleError> {
+        if is_webrtcsink {
+            // WebRTCSink requires no payload.
+            return Ok("".to_string());
+        }
+
+        let pipeline_payload = match &encode {
+            // Here we are naming the payloader as pay0 because the rtsp server
+            // expects this specific name, and having a name doesn't hurt any
+            // other endpoint type.
+            VideoEncodeType::H264 => concat!(
+                " ! h264parse",
+                " ! queue",
+                " ! rtph264pay name=pay0 config-interval=10 pt=96",
+            ),
+            VideoEncodeType::YUYV => concat!(
+                " ! rtpvrawpay name=pay0",
+                // Again, as we are always using the "UYVY" format for raw
+                // application/rtp payloads, "YCbCr-4:2:2" will always be
+                // the right one to pick.
+                " ! application/x-rtp,payload=96,sampling=YCbCr-4:2:2",
+            ),
+            VideoEncodeType::MJPG => " ! rtpjpegpay name=pay0 pt=96",
+            video_encode_type => {
+                return Err(SimpleError::new(format!(
+                    "Unsupported VideoEncodeType: {video_encode_type:#?}"
+                )))
+            }
+        };
+        Ok(pipeline_payload.to_string())
+    }
+
+    fn build_pipeline_sink(
+        endpoints: &Vec<url::Url>,
+        is_webrtcsink: bool,
+        webrtc_name: Option<&String>,
+    ) -> Result<String, SimpleError> {
+        if is_webrtcsink {
+            let (stun_endpoint, turn_endpoint, signalling_endpoint) =
+                Pipeline::build_webrtc_endpoints(&endpoints)?;
+            let capability = "video/x-h264"; // We could also choose for video/x-vp9 here.
+            let webrtc_name = webrtc_name.unwrap();
```

Comment:
Why is this an optional ? Maybe because this is controller also by `is_webrtcsink` ? If so, there are better ways to handle it, like creating another function where the is_webrtcsink logic can exist.